### PR TITLE
enh: improve fetched record buffer types

### PR DESCRIPTION
**Pull request by HaecheonLee**
This PR improves types:

- Create new type that is used in multiple places.
- Set the actual type by giving `FetchedRecordBuffer` instead of `any`.

### CODE REVIEW

File: core/src/fetched-record-buffer.ts [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
 export class FetchedRecordBuffer {
-  store: { [subject: string]: { [position: string]: FetchedRecord[] } }
+  store: SubjectRecords
 
   constructor() {
     this.store = {}
   }
 
-  static fromStore(store: { [subject: string]: { [position: string]: FetchedRecord[] } }) {
+  static fromStore(store: SubjectRecords) {
```
Comment (exAspArk) [inline core/src/fetched-record-buffer.ts:13]: `SubjectRecords` might not be the best name for this complex type. What do you think about reverting this? My preference is to PRY (please repeat yourself) over DRY (don't repeat yourself) in this case until this type is used somewhere externally or needed in more places

File: core/src/fetched-record-buffer.ts [hunk 2]
```diff
@@ -7,14 +7,13 @@ export class FetchedRecordBuffer {
     this.store = {}
   }
 
-  static fromStore(store: { [subject: string]: { [position: string]: FetchedRecord[] } }) {
-    const buffer = new FetchedRecordBuffer()
+  static fromStore(store: { [subject: string]: { [position: string]: FetchedRecord[] } }) {    const buffer = new FetchedRecordBuffer()
```
Comment (exAspArk) [inline core/src/fetched-record-buffer.ts:10]: It looks like this got combined into a single line

File: core/src/fetched-record-buffer.ts (hunk 1 above)
Comment (HaecheonLee) [reply core/src/fetched-record-buffer.ts]: This is updated

File: core/src/fetched-record-buffer.ts (hunk 2 above)
Comment (HaecheonLee) [reply core/src/fetched-record-buffer.ts]: Sorry, I didn't notice. This is updated.

**Pull request merged**